### PR TITLE
Issue 1031 add empty state for sections

### LINKED ISSUE #1031: Add empty state for sections
> As a user, when I see an empty screen I wanna know if it is a bug or there is simply no data to display

To avoid the following situation:

Let's add "small" empty state illustrations for sections. Currently, we only have empty states on full screens.
The following empty state illustration and message has to be implemented everywhere where we have no data to display
❗️: For sections with Call to Actions, keep the Call to Action underneath the illustration

* Mockup:

* Copywriting:
Body 1: "There are no items to display"
Body 2 : "Please check back for updates"

*Image:
Icons

### PR DESCRIPTION
**Pull request by Ekaanth**
fix #1031

### CODE REVIEW

File: src/ui/screens/DemocracyScreen.tsx [hunk 1]
```diff
@@ -65,6 +75,7 @@ export function DemocracyScreen() {
                 </>
               );
             }}
+            renderSectionFooter={groupedDataEmpty}
```
Comment (ishrak-imam) [inline src/ui/screens/DemocracyScreen.tsx]: NIT 💅 

If the logic is implemented directly here we don't have to create extra types in the function above.
Also we can drop the `<Card />` here.

```suggestion
            renderSectionFooter={({section}) =>
              section.data.length < 1 ? (
                <View style={globalStyles.paddedContainer}>
                  <EmptyState subheading="There are no items to display" caption="Please check back for updates" />
                </View>
              ) : null
            }
```

File: src/ui/screens/DemocracyScreen.tsx [hunk 2]
```diff
@@ -65,6 +75,7 @@ export function DemocracyScreen() {
...
             keyExtractor={(item) => item.index.toString()}
             ListEmptyComponent={EmptyView}
```
Comment (ishrak-imam) [inline src/ui/screens/DemocracyScreen.tsx:80]: This is not needed anymore.

File: src/ui/components/EmptyState.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    width: '30%',
+    height: '20%',
+    paddingHorizontal: standardPadding * 2,
+  },
+  emptyContainer: {
+    maxWidth: '90%',
+    paddingHorizontal: standardPadding,
+    flexWrap: 'wrap',
+  },
+});
```
Comment (ishrak-imam) [inline src/ui/components/EmptyState.tsx]: This is a simple reusable component. The idea is to apply the necessary padding/margin where it is used. The requirement of spacing may not be same everywhere.

```suggestion
const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  imageContainer: {
    flex: 1
  },
  emptyContainer: {
    flex: 3,
  },
});
```

File: src/ui/screens/DemocracyScreen.tsx (hunk 2 above)
Comment (ishrak-imam) [inline src/ui/screens/DemocracyScreen.tsx]: `ListEmptyComponent` prop is not needed anymore.

File: src/ui/screens/DemocracyScreen.tsx (hunk 1 above)
Comment (Ekaanth) [reply src/ui/screens/DemocracyScreen.tsx]: We need to have the `Card`, else the  whole background will be white, ref :

File: src/ui/screens/DemocracyScreen.tsx [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import {SectionList, StyleSheet, View, RefreshControl} from 'react-native';
+import {SectionList, StyleSheet, View, RefreshControl, SectionListData} from 'react-native';
```
Comment (ishrak-imam) [inline src/ui/screens/DemocracyScreen.tsx]: 👀   `SectionListData`

**Review by ishrak-imam — APPROVED** [review]
LGTM!!

File: src/ui/components/EmptyState.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import {StyleSheet, Image, View} from 'react-native';
+import EmptyList from 'image/EmptyList.png';
+import {Subheading, Caption} from '@ui/library';
+
+type Props = {
+  subheading: string;
+  caption?: string;
+};
+
+export function EmptyState({subheading, caption}: Props) {
```
Comment (silva-fj) [inline src/ui/components/EmptyState.tsx]: How about `EmptyStateTeaser` ?

File: src/ui/screens/DemocracyScreen.tsx [hunk 6]
```diff
@@ -65,8 +65,14 @@ export function DemocracyScreen() {
                 </>
               );
             }}
+            renderSectionFooter={({section}) =>
+              section.data.length < 1 ? (
+                <Card style={globalStyles.paddedContainer}>
+                  <EmptyState subheading="There are no items to display" caption="Please check back for updates" />
```
Comment (silva-fj) [inline src/ui/screens/DemocracyScreen.tsx]: "There are no items to display" is to generic.. It should be "**There are no active referendums**" and "**There are not active proposals**" accordingly

File: src/ui/screens/TreasuryScreen.tsx [hunk 7]
```diff
@@ -107,7 +108,13 @@ function TreasuryOverviewScreen() {
                 </View>
               );
             }}
-            ListEmptyComponent={EmptyView}
+            renderSectionFooter={({section}) =>
+              section.data.length < 1 ? (
+                <Card style={globalStyles.paddedContainer}>
+                  <EmptyState subheading="There are no items to display" caption="Please check back for updates" />
```
Comment (silva-fj) [inline src/ui/screens/TreasuryScreen.tsx]: Same here, the text is too generic

File: src/ui/components/EventsCalendarTeaser.tsx [hunk 8]
```diff
@@ -35,15 +35,7 @@ export function EventsCalendarTeaser(props: Props) {
           </React.Fragment>
         ))
```
Comment (silva-fj) [inline src/ui/components/EventsCalendarTeaser.tsx:36]: There is an issue with this logic, when first loading the screen (with no data) the loading component is not being displayed, it shows the empty state component for a moment and then shows the upcoming events. Can you please check that as well?

Comment (ishrak-imam) [reply src/ui/components/EventsCalendarTeaser.tsx:36]: I fixed it [here](https://github.com/litentry/litentry-app/pull/1030/commits/c2d0a219a3b13772468c8f84404472185ceebb3c)

File: src/ui/components/EventsCalendarTeaser.tsx [hunk 9]
```diff
@@ -17,8 +17,8 @@ export function EventsCalendarTeaser(props: Props) {
   const {data: events, loading} = useCalendarEvents();
   const topThreeEvents = events?.slice(0, 3) ?? [];
   return (
-    <SectionTeaserContainer onPress={props.onPress} title="Events Calendar">
-      {loading && !events ? (
+    <SectionTeaserContainer onPress={props.onPress} title="Upcoming events">
+      {loading && !topThreeEvents ? (
```
Comment (ishrak-imam) [inline src/ui/components/EventsCalendarTeaser.tsx]: The value of the `topThreeEvents` variable is always truthy. So `!topThreeEvents` will make the condition false every time and loading will never appear.

```suggestion
      {loading && !events ? (
```

Comment (Ekaanth) [reply src/ui/components/EventsCalendarTeaser.tsx]: I should Have kept a close eye on merging. It is resolved now.

**Review by ishrak-imam — APPROVED** [review]

**Pull request merged**